src/Symfony/Component/Intl/Countries.php

```diff
@@ -21,7 +21,7 @@
  */
 final class Countries extends ResourceBundle
 {
-    private static bool $optionalUserAssigned;
+    private static ?bool $allowUserAssigned = null;
```

Comment:
Maybe this is technically required for some reasons, but I'd prefer to not have nullable booleans. Can we make this option truly binary and give it a `false` default value?

Reply:
It should not be initialized by default. So that if it's used directly, an error will occur.

```suggestion
    private static bool $allowUserAssigned;
```

Reply:
when it's null, we populate it from the env var

Reply:
This was so that I could write [this test](https://github.com/symfony/symfony/pull/61024/files#diff-ac1ae93f940a0511a00c36b9093cf58f706ca3708575dd20113ff6498b7a13ceR800) that does in fact "simulate" env var behavior. 

I am also not a fan of the nullable but I could not find a way to write a test that was clear what it was testing. Open to ideas as Roland was the one who wanted that test, and I also agree it is good to have.

Edit: I see the link does not work when UI is collapsed. I meant: 

```php
    public function testEnvVariableAllowsUserAssigned(): void
    {
        // when env param is not set
        $ref = new \ReflectionProperty(Countries::class, 'allowUserAssigned');
        $ref->setValue(null, null);

        $this->assertFalse(Countries::exists('XK'));

        // when env param is set to false
        $ref = new \ReflectionProperty(Countries::class, 'allowUserAssigned');
        $ref->setValue(null, null);

        putenv('SYMFONY_INTL_ALLOW_USER_ASSIGNED=false');

        $this->assertFalse(Countries::exists('XK'));

        // when env param is set to true
        $ref = new \ReflectionProperty(Countries::class, 'allowUserAssigned');
        $ref->setValue(null, null);

        putenv('SYMFONY_INTL_ALLOW_USER_ASSIGNED=true');

        $this->assertTrue(Countries::exists('XK'));
    }
```

Reply:
There is a design issue that leads to use the reflection. The environment variable should not be the only way to define this configuration. We could add a static method to assign the value. 

```php
public static function setWithUserAssigned(bool $withUserAssigned): ?bool
{
    $previous = self::$withUserAssigned;

    self::$withUserAssigned = $withUserAssigned;

    return $previous;
}
```

The class being final, we can change the parameter type later to accept a list of code, we we decides to do so.

Reply:
Maybe you could keep the value as non-nullable and uninitialized, and (unfortunately for perfs) run each tests related to this var in a separate process, so that the var is truly reset every time. It is also closer to what happens in a real execution environment.

Reply:
I am going with GromNaN's recommendation because it is something I considered myself. This could also cover cases when the country list you need to use is tied to a sf-user-account, so for some you have it with-user-assigned and for some without-user-assigned

Reply:
So, I found a way to implement all ideas from all three. 

- the internal property now is `bool` only, not nullable
- there is a public method to change this property at runtime if you wish to do so
- changed GHA a bit 🤏 so I can run the ENV tests in isolation (this can be done easier with PHPUnit 11)

Let me know. But I will have to pick up next Monday, as work calls now :)

Reply:
> changed GHA a bit 🤏 so I can run the ENV tests in isolation (this can be done easier with PHPUnit 11)

cc @xabbuh as you are working on
* #58370